src/libs/TransactionUtils.ts

```diff
@@ -427,7 +428,7 @@ function getValidWaypoints(waypoints: WaypointCollection, reArrangeIndexes = fal
         const previousWaypoint = waypointValues[lastWaypointIndex];
 
         // Check if the waypoint has a valid address
-        if (!waypointHasValidAddress(currentWaypoint)) {
+        if (!waypointHasValidName(currentWaypoint) || !waypointHasValidAddress(currentWaypoint)) {
```

Comment:
The name shouldn't have anything to do with the validation, since it's not a required field. While offline the name won't be filled in. This is similar to how we don't validate lat/lng even though they should be filled in by the frontend when online. On the backend we fill in all these optional fields if needed.

Reply:
@neil-marcellini Oh I see that we mean to treat name as an 'optional' field... I made all the changes with respect to it being a 'required' field. That is the main difference here...
Once again hinges on t[his review comment](https://github.com/Expensify/App/pull/29931#discussion_r1367374834)